chia/wallet/nft_wallet/nft_puzzles.py

```diff
@@ -207,6 +209,53 @@ def get_percentage_from_puzzle(puzzle: Program) -> Optional[uint64]:
     return None
 
 
+def get_nft_info_from_puzzle(puzzle: Program, nft_coin: Coin) -> NFTInfo:
+    """
+    Extract NFT info from a full puzzle
+    :param puzzle: NFT full puzzle
+    :param nft_coin: NFT coin
+    :return: NFTInfo
+    """
+    # TODO Update this method after the NFT code finalized
+    curried_args = match_nft_puzzle(puzzle)[1]
+    (
+        _,
+        singleton_struct,
+        current_owner_did,
+        nft_transfer_program_hash,
+        transfer_program_curry_params,
+        metadata,
+    ) = curried_args
+    (
+        royalty_address,
+        trade_price_percentage,
+        settlement_mod_hash,
+        cat_mod_hash,
+    ) = transfer_program_curry_params.as_iter()
```

Comment:
`transfer_program_curry_params` could be None if `match_nft_puzzle() returns (False, iter(())).

Some sort of error handling would be nice, even if it's just adding an `assert transfer_program_curry_params is not None`. That would at least help point toward the failure in `match_nft_puzzle`.